### ISSUE TITLE
resource/aws_rds_cluster: Terraform 0.14 ignore_changes Handling for replication_source_identifier

### ISSUE THREAD

**Issue by bflad** [hashicorp/terraform-provider-aws#15643]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Future versions of Terraform CLI, starting with 0.14, are more sensitive to how the `lifecycle` configuration block `ignore_changes` argument is applied. Directionally, Terraform's core logic has been moving towards being more explicit about what Terraform Providers can and cannot suggest towards having configurations differ from the final applied state. These differences typically are what cause subtle bugs in core's handling across its whole feature set or represent situations where a generated plan may not exactly match the applied intention of a configuration.

In our Terraform 0.14.0-alpha20201007 testing of the provider, this failure is present:

```
=== CONT  TestAccAWSRDSCluster_GlobalClusterIdentifier_PrimarySecondaryClusters
TestAccAWSRDSCluster_GlobalClusterIdentifier_PrimarySecondaryClusters: resource_aws_rds_cluster_test.go:1355: Step 1/1 error: Error running pre-apply refresh:
Error: Cannot ignore argument not set in the configuration
on config870555763/terraform_plugin_test.tf line 83, in resource "aws_rds_cluster" "secondary":
83:       replication_source_identifier,
The ignore_changes argument is not set in the configuration.
The ignore_changes mechanism only applies to changes within the configuration,
and must be used with arguments set in the configuration and not computed by
the provider.
--- FAIL: TestAccAWSRDSCluster_GlobalClusterIdentifier_PrimarySecondaryClusters (2.84s)
```

This attribute is used for creating an Aurora read replica on Terraform `aws_rds_cluster` resource creation or as in the case of this test, an Aurora global cluster member will automatically have this property managed by AWS. Given the latter situation but the possibility for the first situation if its value was defined in the configuration, the Terraform resource must mark this attribute with `Computed: true` so it can be properly set by the provider without a configuration (and `ignore_changes` omitted). This effectively disables drift detection when not configured, which should be documented during the resource update.

### Affected Resource(s)

* aws_rds_cluster

### Potential Terraform Configuration

```hcl
resource "aws_rds_cluster" "secondary" {
  provider                  = "awsalternate"
  cluster_identifier        = "%[3]s"
  db_subnet_group_name      = aws_db_subnet_group.alternate.name
  skip_final_snapshot       = true
  source_region             = data.aws_region.current.name
  global_cluster_identifier = aws_rds_global_cluster.test.id
  engine                    = aws_rds_global_cluster.test.engine
  engine_version            = aws_rds_global_cluster.test.engine_version
  depends_on                = [aws_rds_cluster_instance.primary]

  # This configuration needs to be removed below with passing tests
  lifecycle {
    ignore_changes = [
      replication_source_identifier,
    ]
  }
}
```

### References

* [Fix handling of ignore_changes](https://github.com/hashicorp/terraform/pull/26387)

**Comment by marinsalinas** [comment]
Hey @bflad, this issue will be fixed before 0.14? In my current use case, I had to add ignore_changes on replication_source_identifier.

**Comment by bflad** [comment]
@marinsalinas great question! We will actually need to double check with Terraform CLI 0.14.0-rc1 whether or not this is still applicable since there was a very related update to `ignore_changes` handling from the prior pre-release. If it still is, we are tracking to have a code fix in by the time of 0.14.0 final. 👍

**Comment by bflad** [comment]
The acceptance test that previously failed in the early alpha build is passing with rc1, so my hunch here is that nothing will be required to do before the Terraform 0.14 release with the resource.
